tnt/src/main/java/org/dcsa/conformance/standards/tnt/checks/TntChecks.java

```diff
@@ -0,0 +1,389 @@
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import static org.dcsa.conformance.standards.tnt.checks.TntDataSets.VALID_DOCUMENT_TYPE_CODES;
+import static org.dcsa.conformance.standards.tnt.checks.TntDataSets.VALID_EQUIPMENT_EVENT_TYPES;
+import static org.dcsa.conformance.standards.tnt.checks.TntDataSets.VALID_EVENT_TYPES;
+import static org.dcsa.conformance.standards.tnt.checks.TntDataSets.VALID_SHIPMENT_EVENT_TYPES;
+
+@UtilityClass
+public class TntChecks {
+  private static final String SHIPMENT_EVENT_TYPE = "SHIPMENT";
+  public static ActionCheck responseContentChecks(UUID matched, String standardVersion, Supplier<SuppliedScenarioParameters> sspSupplier) {
+
+    var checks = new ArrayList<JsonContentCheck>();
+
+    checks.add(JsonAttribute.customValidator(
+      "Validate eventIds are unique",
+      body -> {
+        Set<String> uniqueEventIds = StreamSupport.stream(body.spliterator(), false)
+          .map(node -> node.path("eventID").asText())
+          .collect(Collectors.toSet());
+          if(uniqueEventIds.size() != body.size()) {
+            return Set.of("Event Ids are not unique");
+          }
+        return Set.of();
+      }
+    ));
+
+    checks.add(JsonAttribute.customValidator(
+      "Validate eventCreatedDateTime parameter conditions are met",
+      body -> {
+        Set<OffsetDateTime> eventCreatedDateTimes = StreamSupport.stream(body.spliterator(), false)
+          .map(node -> OffsetDateTime.parse(node.path("eventCreatedDateTime").asText()))
+          .collect(Collectors.toSet());
+        Optional<Map.Entry<TntFilterParameter, String>> eventCreatedDateTimeParam = sspSupplier.get().getMap().entrySet().stream()
+          .filter(e -> e.getKey().equals(TntFilterParameter.EVENT_CREATED_DATE_TIME))
+          .findFirst();
+        if (eventCreatedDateTimeParam.isPresent()) {
+          String[] eventCreatedDateTimeParamKey = eventCreatedDateTimeParam.get().getKey().getQueryParamName().split(":");
+          OffsetDateTime eventCreatedDateTimeParamValue = OffsetDateTime.parse(eventCreatedDateTimeParam.get().getValue());
+          if (eventCreatedDateTimeParamKey.length == 2) {
+            return validateEventCreatedDateTime(eventCreatedDateTimes, eventCreatedDateTimeParamKey, eventCreatedDateTimeParamValue);
+          }
+        }
+        return Set.of();
+      }
+    ));
+
+    checks.add(JsonAttribute.customValidator(
+      "Validate eventTypes are valid",
+      body -> {
+        Set<String> invalidEventTypes = StreamSupport.stream(body.spliterator(), false)
+          .map(node -> node.path("eventType").asText())
+          .filter(eventType -> !VALID_EVENT_TYPES.contains(eventType))
+          .collect(Collectors.toSet());
+
+        if (!invalidEventTypes.isEmpty()) {
+          return invalidEventTypes.stream()
+            .map(eventType -> "Invalid eventType: " + eventType)
+            .collect(Collectors.toSet());
+        }
+        return Set.of();
```

Comment:
What happens if you remove the `if (!invalidEventTypes.isEmpty()) {` line and just stream over the empty Set? It also returns an empty Set, I would assume, which is fine right? If that works, just remove line 83. Same for the other checks below and above.

Reply:
the error message needs to be provided back to the subChecks, hence i need to have it.